src/object/moving_sprite.hpp

```diff
@@ -53,11 +53,11 @@ class MovingSprite : public MovingObject
 
   virtual ObjectSettings get_settings() override;
   virtual void after_editor_set() override;
-  virtual void on_type_change(int old_type = -1) override;
+  virtual void on_type_change(int old_type) override;
```

Comment:
Technically, you can leave the default. IDK...

Reply:
The whole semantic of that function is...weird

Reply:
Removed the default so someone editing the code can't accidentally call it with `old_type` as `-1`, which it will detect as an initial call.

Reply:
Pain!

Could you not add an m_type property to MovingObject, initialize that to -1 and check against that, then pass the current type as a value?

Semantically, this would make more sense.

Reply:
You mean `MovingSprite`? Honestly, don't think adding another variable just for this, which is a copy of `GameObject::m_type` would make sense, neither would it be better for memory. What I'm thinking of is I could add a comment somewhere indicating values for `old_type`, which are below 0, should only be passed on object initialization?

Reply:
Can we check against m_type?

Reply:
Another solution is to maybe initialize `GameObject::m_type` as -1 and then in `parse_type` set that to 0? Though that would also be a problem, since some objects call `parse_type` multiple times, because of inheritance.